### DISCUSSION TITLE
Data coming from Livewire  reactive in Vue

### DISCUSSION THREAD

**Discussion by renatokira** [livewire/livewire#5571] [1. Help] [answered]
Hello everyone!

I'm working on an application that uses Livewire and Vue, and I would like to work reactively to access data sent from Livewire to the Vue component. Currently, I am using the following code to get the data:

```javascript

<template>
    <div>
        <h1>Wire Data</h1>
        <pre>{{ wireData }}</pre>
    </div>
</template>

<script>
export default {
    data() {
        return {
            componentWire: null,
            wireData: []

        }
    },
    mounted() {
        console.log('Mounted TestWire');
        Livewire.hook('component.initialized', (component) => {
            if (component.name == 'chamados.imoc.pages.rotas-list') {
                this.componentWire = component;
                this.wireData = this.componentWire?.data;

            }
        });

        Livewire.hook('message.processed', (el, component) => {
            this.wireData = this.componentWire?.data;
            console.log(this.componentWire?.data);
        })
    }
};
</script>
```

However, I feel that this approach may not be the best way to work with reactive data in Livewire and Vue. I would like to know if there is a better way to work with reactive data in this context.

I would like to understand how to work with reactive data in Livewire and Vue in a more efficient and elegant way. My goal is to be able to update and manipulate data sent from Livewire to the Vue component in real-time.

Using [Livewire VueJS Support Plugin](https://github.com/livewire/vue) to use VueJS and Livewire on the same page

![image](https://user-images.githubusercontent.com/10859156/220463902-626d6a38-3908-473d-8d96-042ee5ae55b8.png)

**Comment by joshhanley** [comment, marked as answer]
@renatokira I'm not sure I follow exactly, it seems you have worked out how to access Livewire's data. Is there something wrong with that? Is it not working?

**Reply by renatokira** [reply]
Hi,@joshhanley I would like to know if there are other approaches to reactivity for Livewire and Vue, besides what I'm trying to implement above. To be honest, I don't have much experience with Livewire.

**Reply by joshhanley** [reply]
@renatokira personally I wouldn't use Vue and Livewire together. While it can work, there can also be some strange bugs with it. I've no experience using them together. So if you can get them to work in a way that suits you, then I'd say use that, as there's no standard way of doing it or any best practices for using them together.

But one thing you can look into, is how Livewire and Alpine work together (they were designed to work together though), it may give you some more ideas of what _might_ be possible between Livewire and Vue.

Is there any reason you are using both? If your app is all Vue, I would suggest looking at InertiaJS as the interface between Laravel and Vue.
